proposals/dictionary-expressions.md

```diff
@@ -165,47 +165,52 @@ From [Open Question 3](#open-question-3), we will support *dictionary elements*
 2. Use applicable instance indexer if available; otherwise report an error during construction (or conversion?).
 3. Use C#12 initialization always.
 
+## Dictionary types
+
+A type is considered a *dictionary type* if the following hold:
+* The *element type* is `KeyValuePair<TKey, TValue>`.
+* The *type* has an instance *indexer* where:
+  * The indexer has a single parameter with an identity conversion from the parameter type to `TKey`.\*
+  * There is an identity conversion from the indexer type to `TValue`.\*
+  * The getter returns by value.
+  * The getter is as accessible as the declaring type.
```

Comment:
thanks! i really like this. it seems like it will match intuitions a lot on what a dictionary actually is.